test/integration/cdi/example-driver/app/kubeletplugin.go

```diff
@@ -0,0 +1,153 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"k8s.io/klog/v2"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+	plugin "k8s.io/kubernetes/test/integration/cdi/example-driver/kubeletplugin"
+)
+
+const (
+	deviceName = "exampledevice"
+	cdiVersion = "0.2.0"
+)
+
+type examplePlugin struct {
+	cdiDir                 string
+	driverName             string
+	draAddress             string
+	pluginRegistrationPath string
+}
+
+// newExamplePlugin returns an initialized examplePlugin instance.
+func newExamplePlugin(config examplePlugin) (*examplePlugin, error) {
+	return &examplePlugin{
+		cdiDir:                 config.cdiDir,
+		driverName:             config.driverName,
+		draAddress:             config.draAddress,
+		pluginRegistrationPath: config.pluginRegistrationPath,
+	}, nil
+}
+
+// getJsonFilePath returns the absolute path where json file is/should be.
+func (ex *examplePlugin) getJsonFilePath(claimUID string) string {
+	jsonfileName := fmt.Sprintf("%s-%s.json", ex.driverName, claimUID)
+	return filepath.Join(ex.cdiDir, jsonfileName)
+}
+
+// startPlugin starts servers that are necessary for plugins and registers the plugin with kubelet.
+func (ex examplePlugin) startPlugin() error {
+	klog.Infof("Starting kubelet plugin")
+
+	// create CDI directory if not exists
+	if err := os.MkdirAll(ex.cdiDir, os.FileMode(0750)); err != nil {
+		return fmt.Errorf("failed to create directory: %v", err)
+	}
+
+	// create DRA driver socket directory if not exists.
+	if err := os.MkdirAll(filepath.Dir(ex.draAddress), 0750); err != nil {
+		return fmt.Errorf("failed to create DRA driver socket directory: %v", err)
+	}
+
+	// Run a grpc server for the driver, that listens on draAddress
+	if err := plugin.StartNonblockingGrpcServer(ex.draAddress, &ex); err != nil {
+		return fmt.Errorf("failed to start a nonblocking grpc server: %v", err)
+	}
+
+	// Run a registration server and registers plugin with kubelet
+	if err := plugin.StartRegistrar(ex.driverName, ex.draAddress, ex.pluginRegistrationPath); err != nil {
+		return fmt.Errorf("failed to start registrar: %v", err)
+	}
+
+	return nil
+
+}
+
+func (ex *examplePlugin) NodePrepareResource(ctx context.Context, req *drapbv1.NodePrepareResourceRequest) (*drapbv1.NodePrepareResourceResponse, error) {
+
+	kind := ex.driverName + "/" + deviceName
+
+	klog.Infof("NodePrepareResource is called: request: %+v", req)
+
+	klog.Infof("Creating CDI Files")
```

Comment:
Do we create multiple CDI files?

Reply:
I, from the begining, asked multiple times
1. why you say the plugin generates CDI fileS though we create only JSON file.
2. Is there any other file I need to generate?

I got no answer, and kept it like that?

Reply:
There is only one file.

Reply:
Yes, changed.